addon/validations/factory.js

```diff
@@ -75,30 +76,27 @@ function setOwner(obj, model) {
  * @return {Ember.Mixin}
  */
 export default function buildValidations(validations = {}) {
-  var validatableAttrs = Object.keys(validations);
-  var props = createGlobalValidationProps(validatableAttrs);
-  var attrs = {};
-
-  // Private
-  props._validators = {};
-  props._debouncedValidations = {};
-  props._validatableAttributes = validatableAttrs;
-  props._validationRules = validations;
-
   processDefaultOptions(validations);
 
-  validatableAttrs.forEach((attribute) => {
-    attrs[attribute] = createCPValidationFor(attribute, validations[attribute]);
-  });
+  const Validations = createValidationsObject(validations);
 
-  var AttrValidations = Ember.Object.extend(attrs);
-  var GlobalValidations = Ember.Object.extend(props, {
-    isValidations: true,
-    validate,
-    validateSync
-  });
+  return Ember.Mixin.create({
+    validations: null,
 
-  return createMixin(GlobalValidations, AttrValidations);
+    init() {
+      this._super(...arguments);
+      this.set('validations', Validations.create({
```

Comment:
That's what it was before, but because of [this line](https://github.com/offirgolan/ember-cp-validations/pull/112/files#diff-795f2437a28d93a92495ebda9b421065R157), it goes into an infinite loop.